### PR TITLE
Replace git apply with patch

### PR DESCRIPTION
**Pull request by maxim-h**
Hey, thanks a lot for the AUR package.
Unfortunately I've not been able to install it.
After some manual tinkering with `makepkg` I realized that `git apply` actually just skips both patch files because it searches for them from the root of the git repo instead of the current working directory. The same issue is discussed [here](https://stackoverflow.com/questions/42386491/git-apply-skips-patches). I don't really understand why I had this error and you did not. I have `git version 2.34.0`.
The easiest way to fix it for me was to just not use `git` for that and replace it with `patch`.
I confirmed that it builds and installs normally after that.

### CODE REVIEW

**Comment by trap000d** [comment]
Hmm. Just tested on random Arch machine  - ```git apply``` works well both either with manual build (```makepkg```) or with `yay rstudio-desktop`
```
me@home:~$ git --version
git version 2.34.0
```
I used git apply here because in the past had issues with patch files created by ```git diff```. For unclear reasons (trailing whitespace?) sometimes ```patch``` couldn't apply them at all, whereas ```git apply``` worked.

Anyway, thanks for PR. Rstudio is pretty massive package so I'd not like to create a new release - not everyone have 64-core-256GiB machine to build packages every several days :). 
I'll mention your PR in aur - if any other users have such problem, then I'll make updates. Otherwise I'll implement them for next Rstudio release

**Comment by trap000d** [comment]
Just few more thoughts about your build environment.
```makepkg``` on prepare() must create symlinks from all source files to build directory. Here is how it looks on my computer  right after ```makepkg -o```
```shell
me@home:/var/tmp/makepkg/rstudio-desktop/src$ ls -la
total 32
drwxr-xr-x  5 me me 4096 Nov 24 14:11 .
drwxr-xr-x  3 me me 4096 Nov 24 14:11 ..
-rw-rw-r--  1 me me  693 Nov 24 14:11 cran_multithread.patch
lrwxrwxrwx  1 me me   51 Nov 24 14:11 gin-2.1.2.zip -> /home/me/src/rs/rstudio-desktop/gin-2.1.2.zip
drwxr-xr-x  4 me me 4096 Oct 18  2017 gwt-2.8.2
lrwxrwxrwx  1 me me   51 Nov 24 14:11 gwt-2.8.2.zip -> /home/me/src/rs/rstudio-desktop/gwt-2.8.2.zip
drwxr-xr-x  6 me me 4096 Aug 11 13:48 node-v14.17.5-linux-x64
lrwxrwxrwx  1 me me   68 Nov 24 14:11 node-v14.17.5-linux-x64.tar.gz -> /home/me/src/rs/rstudio-desktop/node-v14.17.5-linux-x64.tar.gz
lrwxrwxrwx  1 me me   56 Nov 24 14:11 node_version.patch -> /home/me/src/rs/rstudio-desktop/node_version.patch
lrwxrwxrwx  1 me me   45 Nov 24 14:11 qt.conf -> /home/me/src/rs/rstudio-desktop/qt.conf
drwxr-xr-x 13 me me 4096 Nov  9 11:04 rstudio-2021.09.1-372
lrwxrwxrwx  1 me me   67 Nov 24 14:11 rstudio-2021.09.1.+372.tar.gz -> /home/me/src/rs/rstudio-desktop/rstudio-2021.09.1.+372.tar.gz

```


I've got the following in my /etc/makepkg.conf
```
BUILDDIR=/var/tmp/makepkg
```

**Comment by maxim-h** [comment]
Ok, now it makes more sense to me.

I think the difference was that I never had the `BUILDDIR` set in my `/etc/makepkg.conf` (it was commented out). But after uncommenting it `yay` managed to install `rstudio-desktop` from **aur**. 

So what happened was that when `BUILDDIR` is not set the source files were copied by `yay` into my `$XDG_CACHE_HOME/yay/` which happens to be `$HOME/.cache/yay`. But because I use `git` for my dotfiles the whole `$HOME` is a huge `git` repo (with almost everything ignored of course). And obviously when I tested with `makepkg` they were within your *aur* repo. So in this scenario `git apply` interprets all the paths in the `.patch` files relative to the root of the repo and skips them all.

On the other hand when `BUILDDIR=/var/tmp/makepkg` there's no `git` repo "around" it, so the only way `git apply` can interpret the paths is relative to the working directory which is the expected behavior in your PKGBUILD. And in this case everything just works. 

TL;DR: `git apply` behavior is reasonable, but not intuitive and not very well documented. If your `BUILDDIR` happens to be inside a `git` repo, which is the case for people who use all default parameters AND have a `.git` in their `$HOME`, `git apply` will skip the patches. 
Not sure how to fix it exactly while keeping `git apply`. You always need to know where your `BUILDDIR` is located relative to the closest `git` root (as mentioned [here](https://stackoverflow.com/questions/42386491/git-apply-skips-patches) absolute paths don't work either).
